src/inhibitors/requiredSettings.js

```diff
@@ -3,10 +3,17 @@ const { Inhibitor } = require('klasa');
 module.exports = class extends Inhibitor {
 
 	async run(message, command) {
-		if (!command.requiredSettings.length || message.channel.type !== 'text') return;
-		// eslint-disable-next-line eqeqeq
-		const requiredSettings = command.requiredSettings.filter(setting => message.guild.settings.get(setting) == null);
-		if (requiredSettings.length) throw message.language.get('INHIBITOR_REQUIRED_SETTINGS', requiredSettings);
+		const missingSettings = [];
+
+		for (const [gateway, settings] of Object.entries(command.requiredSettings)) {
+			if (gateway === 'guild' && message.channel.type !== 'text') continue;
```

Comment:
This is an opinion... If a command requires guild settings, then it should alert the user the settings don't exist just like any other missing settings.